From: Deming Wang <wangdeming@inspur.com>
Subject: mm/mincore.c: use vma_lookup() instead of find_vma()
Date: Thu, 6 Oct 2022 23:03:45 -0400

Using vma_lookup() verifies the start address is contained in the found
vma.  This results in easier to read the code.

Link: https://lkml.kernel.org/r/20221007030345.5029-1-wangdeming@inspur.com
Signed-off-by: Deming Wang <wangdeming@inspur.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mincore.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/mincore.c~mm-use-vma_lookup-instead-of-find_vma
+++ a/mm/mincore.c
@@ -190,8 +190,8 @@ static long do_mincore(unsigned long add
 	unsigned long end;
 	int err;
 
-	vma = find_vma(current->mm, addr);
-	if (!vma || addr < vma->vm_start)
+	vma = vma_lookup(current->mm, addr);
+	if (!vma)
 		return -ENOMEM;
 	end = min(vma->vm_end, addr + (pages << PAGE_SHIFT));
 	if (!can_do_mincore(vma)) {
_
